libnmstate/nm/connection.py

```diff
@@ -156,6 +156,10 @@ def _safe_activate_async(self):
 
         specific_object = None
         user_data = cancellable
+
+        logging.debug(
```

Comment:
Why this additional logging is needed?
We have a log when the _safe_activate_async method is called, when it exits early and when the callback is called.
What scenario required this additional log?

Reply:
When activating failed without udev, there was no log message without this about how nmstate tried to activate something (i.e. by device or by name or both). This log message ensures that this is logged.

Reply:
We are logging the execution of this method, but without the arguments:
https://github.com/nmstate/nmstate/blob/master/libnmstate/nm/nmclient.py#L88

At some time in the past the argument logging were removed because they seemed not useful and cause a lot of "noise" in the log.

With this logging addition, a duplication is created and it does not confirm with the other async methods/functions logging. On the other hand, I understand you have been missing a piece of information.

My suggestion is to solve this through the timeout cancellation. In case a timeout occurs or the execution of the callback function is cancelled in general, log the relevant info.
 For this specific case, have you seen a log like the one at line 181? I would expect that in case the udev is not operational, a timeout occurs and the CB is cancelled, outputting an error log.

Reply:
Only logging the activation type on errors does not allow to compare successful runs with broken runs because the successful runs do not log the full information. Are the previous commits ready to be merged? I can open a new PR to get at least them merged.

Reply:
But it is logged on success as well.. check the callback method registered.